### PR TITLE
Add optimistic updates for aggregate operations

### PR DESCRIPTION
**Pull request by sashaluxdev**
## Summary
- Aggregate operation changes (table footer and kanban view) now update the UI immediately instead of waiting for the API response
- On failure, state rolls back to the value visible before the action
- A request-ID counter discards stale responses when rapid successive changes race (edge-case handling).

## Test plan
- [ ] Change aggregate operation on a table column footer — UI updates instantly
- [ ] Simulate API failure (e.g. disconnect network) — UI rolls back to previous value
- [ ] Change aggregate operation on kanban view — same instant update behavior

### CODE REVIEW

File: packages/twenty-front/src/modules/views/hooks/useUpdateViewAggregate.ts [hunk 1]
```diff
@@ -18,6 +21,29 @@ export const useUpdateViewAggregate = () => {
...
+    recordIndexGroupAggregateFieldMetadataItemComponentState,
+  );
+
+  // oxlint-disable-next-line twenty/no-state-useref
+  const latestRequestIdRef = useRef(0);
+  // oxlint-disable-next-line twenty/no-state-useref
+  const lastConfirmedStateRef = useRef({
+    operation: recordIndexGroupAggregateOperation,
+    fieldMetadataItem: recordIndexGroupAggregateFieldMetadataItem,
+  });
```
[the remark this answers is not in the document]
Comment (sashaluxdev) [reply packages/twenty-front/src/modules/views/hooks/useUpdateViewAggregate.ts:47]: This ref is component-scoped and re-initializes on mount. View switches change the component context, causing a remount. The first successful operation after any external state change also updates the ref. Not a practical concern.

File: packages/twenty-front/src/modules/views/hooks/useUpdateViewAggregate.ts [hunk 2]
```diff
@@ -44,6 +70,16 @@ export const useUpdateViewAggregate = () => {
...
       }
 
+      const requestId = ++latestRequestIdRef.current;
+
+      const newFieldMetadataItem =
+        objectMetadataItem.fields?.find(
+          (field) => field.id === kanbanAggregateOperationFieldMetadataId,
+        ) ?? null;
+
+      setRecordIndexGroupAggregateOperation(kanbanAggregateOperation);
```
[the remark this answers is not in the document]
Comment (sashaluxdev) [reply packages/twenty-front/src/modules/views/hooks/useUpdateViewAggregate.ts:82]: The component remounts on view switch (atoms are component-scoped), so the ref resets to correct values for each view. Restoring the pre-update snapshot or syncing via useEffect would reintroduce the original bug by capturing optimistic (unconfirmed) state as the rollback target.

File: packages/twenty-front/src/modules/object-record/record-table/record-table-footer/hooks/useViewFieldAggregateOperation.tsx [hunk 3]
```diff
@@ -42,12 +66,24 @@ export const useViewFieldAggregateOperation = () => {
...
+
+      if (requestId !== latestRequestIdRef.current) {
+        return;
+      }
+    } else {
+      if (requestId !== latestRequestIdRef.current) {
+        return;
+      }
+
+      setViewFieldAggregateOperation(lastConfirmedOperationRef.current ?? null);
```
[the remark this answers is not in the document]
Comment (sashaluxdev) [reply packages/twenty-front/src/modules/object-record/record-table/record-table-footer/hooks/useViewFieldAggregateOperation.tsx:84]: Valid edge case but not practically exploitable — it requires the later request to fail and resolve before the earlier one succeeds. The consequence is a temporary UI mismatch that self-corrects on navigation. Fixing it cleanly requires in-flight request tracking that adds disproportionate complexity, and naive fixes (setting UI on stale success) introduce flicker in the more common concurrent-success case.

**Review by prastoin — CHANGES_REQUESTED** [review]
Hello there thanks for your contribution
Though it does not follow twenty code standards
Especially ref and state etc

**Comment by prastoin** [comment]
Please feel free to open a new one if you want to or pick another easier issue
